modules/editor-events/assets/js/editor/module.js

```diff
@@ -0,0 +1,48 @@
+import EventsStorage from './events-storage';
+
+export default class extends elementorModules.Module {
+	types = {
+		click: 'click',
+	};
+
+	events = {
+		site_settings: 'site_settings',
+	};
+
+	onInit() {
+		if ( ! elementor.config.editor_events?.can_send_events ) {
+			return;
+		}
+
+		window.addEventListener( 'unload', this.sendEvents() );
+	}
+
+	dispatchEvent( type, eventId, context ) {
+		if ( ! elementor.config.editor_events?.can_send_events ) {
+			return;
+		}
+
+		EventsStorage.set( { type, event_id: eventId, context, timestamp: Date.now() } );
+	}
+
+	sendEvents() {
+		const events = EventsStorage.get();
+
+		if ( ! events.length ) {
```

Comment:
In the task I see the following line:
`The system will store and send 100 events each time`

Where is the length limit set?

Reply:
yeah, we discussed with @louiswol94 that we can go in a bit simplier way - just sending 1-by-1 using beacon or in batches on unload. 100 it's not a strict requirement and actually we don't know requirements for now - since data events server is still in development.